lib/puppet/module_tool/applications/builder.rb

```diff
@@ -40,6 +41,22 @@ def archive_file
         File.join(@pkg_path, "#{metadata.release_name}.tar.gz")
       end
 
+      def minitar_verify
+        begin
+          # assume valid if minitar inaccessible (i.e. immediate rescue loading gem)
+          require 'archive/tar/minitar'
+        rescue
+          return true
+        end
+
+        # minitar present, so assume invalid
+        files = Puppet::ModuleTool::Tar::Mini.new.entries(archive_file)
+        valid = files.all? { |f| f.full_name == f.name }
+        if !valid
+          raise Puppet::ModuleTool::Errors::ModuleToolError, "Archive contains invalid entries - may fail minitar extraction on Windows."
```

Comment:
I really want this to be a warning locally, but an error in CI.  Perhaps this needs to be controlled by a global setting.